client/src/Models/Ringdown.js

```diff
@@ -7,20 +7,70 @@ import * as metadata from '../../../shared/metadata';
 import convertToPropType from '../utils/convertToPropType';
 import DeliveryStatus from '../../../shared/constants/DeliveryStatus';
 
+// import ambulance from '../../../shared/metadata/ambulance';
+// import patient from '../../../shared/metadata/patient';
+
+const Patient = metadata.patient.getFieldHash();
+const Ambulance = metadata.ambulance.getFieldHash();
+const EmergencyMedicalServiceCall = metadata.emergencyMedicalServiceCall.getFieldHash();
+
 // define the fields that must all have valid input to make the ringdown valid.  the second array item is an optional function to determine
 // whether the field's current value is valid as input.  by default, the field is counted as having input if its value is truthy.   the
 // array order should be the same as the field order in PatientFields.
+
+const handleInputValidation = (name, value) => {
+  const { type, required = false, allowNull = true } = Patient[name] || Ambulance[name] || EmergencyMedicalServiceCall[name] || {};
+
+  let isValidType;
+  switch (type) {
+    case 'integer':
+      isValidType = typeof value === 'number' && value % 1 === 0;
+      break;
+    case 'boolean':
+      isValidType = typeof value === 'boolean';
+      break;
+    case 'enum':
+    case 'text':
+      isValidType = typeof value === 'string';
+      break;
+    case 'decimal':
+      isValidType = typeof value === 'number';
+      break;
+    default:
+      break;
```

Comment:
Does lint complain if there's no `default`?  Seems unnecessary.  Or if we have one, throw an error from it if it should never be reached.

Reply:
I was actually getting a warning in my console expecting a default, and this fixed that warning. I tried removing and running linter, and it gave me the same warning.

Reply:
Yeah, the current eslint rules seem to require that.  Annoying.  If the default case should never be reached, let's throw an error so it actually earns its keep.

Reply:
Sounds good. Would a  `console.error('error message here')` suffice, or should I return an Error object?

Reply:
I'd just throw an error, as getting there means there's a logic bug, if I understand it correctly.